app/controllers/api/v1/vacations_controller.rb

```diff
@@ -7,6 +7,10 @@ def index
     respond_with @vacations
   end
 
+  def types
```

Comment:
Rails - is a framework for RESTful applications. One of the REST principles is simple CRUD resources.
The way you did it is not bad, but it is not fully REST. I would suggest creating a separate `resources :types` in `vacations` namespace.